lib/spanish_vat_validators.rb

```diff
@@ -24,12 +24,12 @@ def validate_nif(v)
     def validate_cif(v)
       return false if v.nil? || v.empty?
       value = v.clone
-      return false unless value.match(/^[a-wyz][0-9]{7}[0-9a-z]$/i)
+      return false unless value.match(/\A[a-wyz][0-9]{7}[0-9a-z]\z/i)
       even = 0
       odd = 0
       uletter = ["J", "A", "B", "C", "D", "E", "F", "G", "H", "I"]
       text = value.upcase
-      regular = /^[ABCDEFGHJKLMNPQRSVW]\d{7}[0-9,A-J]$/#g);
```

Comment:
`/g` does not exist in ruby. It wasn't doing anything since the `#` converted it into a comment, but if the `#` wasn't there it would probably be an error since `/g` is not an available option. To use global regex searches in ruby you want to use `scan` instead of `match`, but I'm not really sure global searches is ideal here: you want to validate the full string is a NIF, not if the string contains a NIF anywhere. I removed it to make it more clear, because otherwise it could be confusing.